### PR TITLE
rgw/notification: Publish notification information in bucket stats

### PR DESCRIPTION
**Pull request by kchheda3**
Publish notification topics subscribed by bucket in the bucket stats output
As part of notification discussion (https://gist.github.com/yuvalif/15af72d2abf9493cc83f56b94ae5f9cb), it was decided to publish notificaiton information on bucket stats.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/driver/rados/rgw_bucket.cc [hunk 1]
```diff
@@ -1,18 +1,17 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab ft=cpp
 
-#include "rgw_acl_s3.h"
```
Comment (yuvalif) [inline src/rgw/driver/rados/rgw_bucket.cc:4]: why did you change the includes?

File: src/rgw/driver/rados/rgw_bucket.cc [hunk 2]
```diff
@@ -862,11 +861,9 @@ int RGWBucketAdminOp::sync_bucket(rgw::sal::Driver* driver, RGWBucketAdminOpStat
 }
 
 static int bucket_stats(rgw::sal::Driver* driver,
-			const std::string& tenant_name,
-			const std::string& bucket_name,
-			Formatter *formatter,
-                        const DoutPrefixProvider *dpp, optional_yield y)
-{
+                        const std::string& tenant_name,
```
Comment (yuvalif) [inline src/rgw/driver/rados/rgw_bucket.cc:864]: please avoid reformatting unchanged code - this is creating merge conflicts

**Review by yuvalif — APPROVED** [review]

File: src/rgw/driver/rados/rgw_bucket.cc (hunk 2 above)
Comment (kchheda3) [reply src/rgw/driver/rados/rgw_bucket.cc:864]: > please avoid reformatting unchanged code - this is creating merge conflicts

Sorry for the inconvenience.

File: src/rgw/driver/rados/rgw_bucket.cc (hunk 1 above)
Comment (yuvalif) [reply src/rgw/driver/rados/rgw_bucket.cc:4]: please revert the changes in the includes, unless they are needed for the new functionality.

Comment (kchheda3) [reply src/rgw/driver/rados/rgw_bucket.cc:4]: > please revert the changes in the includes, unless they are needed for the new functionality.

You mean I should revet the reordering of #includes and just add the newly added “rgw.pubsub.h” at the end ?

Comment (yuvalif) [reply src/rgw/driver/rados/rgw_bucket.cc:4]: yes please

**Review by yuvalif — APPROVED** [review]

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
teuthology passing: http://pulpito.front.sepia.ceph.com/yuvalif-2023-09-11_12:35:49-rgw:notifications-wip-yuval-test-distro-default-smithi/
